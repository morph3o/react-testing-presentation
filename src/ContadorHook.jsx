import React, { useState } from "react";

const ContadorHook = () => {
  const [contador, setContador] = useState(0);

  return (
    <div>
      <h2>Contador Hook</h2>
      <h3 data-testid="contador">{contador}</h3>
      <button onClick={() => setContador(contador + 1)}>+1</button>
    </div>
  );
};

export default ContadorHook;
