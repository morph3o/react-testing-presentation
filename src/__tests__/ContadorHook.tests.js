import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import ContadorHook from "../ContadorHook";

describe("ContadorHook tests", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ContadorHook />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("should match snapshot", () => {
    const { container } = render(<ContadorHook />);
    expect(container).toMatchSnapshot();
  });

  it("Counter loads with initial state of 0", () => {
    const { container } = render(<ContadorHook />);
    const countValue = getByTestId(container, "counter-text");
    expect(countValue.textContent).toBe("0");
  });

  it("Increment and decrement buttons work", () => {
    const { container } = render(<ContadorHook />);
    const countValue = getByTestId(container, "counter-text");
    const increment = getByTestId(container, "button-increment");

    expect(countValue.textContent).toBe("0");
    fireEvent.click(increment);
    expect(countValue.textContent).toBe("1");
    fireEvent.click(increment);
    expect(countValue.textContent).toBe("2");
  });
});
