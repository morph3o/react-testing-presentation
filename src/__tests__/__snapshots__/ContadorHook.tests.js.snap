// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContadorHook tests should match snapshot 1`] = `
<div>
  <div>
    <h2>
      Contador Hook
    </h2>
    <h3
      data-testid="counter-text"
    >
      0
    </h3>
    <button
      data-testid="button-increment"
    >
      +1
    </button>
  </div>
</div>
`;
